core_fw: Add remote_radio firmware code (#12588)
nrfconnect/sdk-nrf: core_fw/ipc_radio/README.rst

.. _ipc_radio:

> ipc_radio
> #########

Reviewer: ```suggestion
IPC radio firmware
##################
```

.. contents::
   :local:
   :depth: 2

> The ipc_radio firmware allows to use the radio peripheral from another core in a multicore device.

Reviewer: ```suggestion
The IPC radio firmware allows to use the radio peripheral from another core in a multicore device.
```

Requirements
************

The firmware supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340dk_nrf5340_cpunet, thingy53_nrf5340_cpunet

The firmware requires the usage of the :ref:`sysbuild`.

Overview
********

You can use this firmware as a general serialized radio peripheral.

> The firmware can support both Bluetooth Low Energy and IEEE 802.15.4 simultaneously.

Reviewer: ```suggestion
The firmware supports both Bluetooth Low Energy and IEEE 802.15.4 simultaneously.
```

The Bluetooth Low Energy serialization is selectable as HCI or RPC serialization interface.

HCI IPC serialization
=====================

The firmware exposes the :ref:`bluetooth_controller` support to another core using the IPC subsystem.

Host for nRF RPC Bluetooth Low Energy
=====================================

The firmware is running the full Bluetooth Low Energy stack.
It receives serialized function calls that it decodes and executes, then sends response data to the client.

IEEE 802.15.4
=============

> The firmware exposes the the radio driver support to another core using the IPC subsystem.

Reviewer: ```suggestion
The firmware exposes radio driver support to another core using the IPC subsystem.
```

Configuration
*************

> The firmware is enable by using the Sysbuild configuration of :kconfig:option:`SB_CONFIG_NRF_DEFAULT_IPC_RADIO`.

Reviewer: ```suggestion
To enable the firmware, use the Sysbuild configuration of :kconfig:option:`SB_CONFIG_NRF_DEFAULT_IPC_RADIO`.
```

> Supported radio configurations are configured by the following (Sysbuild) Kconfig options:

Reviewer: ```suggestion
You can set the supported radio configurations using the following (Sysbuild) Kconfig options:
```

* :kconfig:option:`SB_CONFIG_NETCORE_IPC_RADIO_BT_HCI_IPC`
* :kconfig:option:`SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC`
* :kconfig:option:`SB_CONFIG_NETCORE_IPC_RADIO_IEEE802154`

> The configurations can be combined to have combined functionalities ie. HCI IPC + IEEE 802.15.4.

Reviewer: ```suggestion
Using these configurations you can combine functionalities, for example HCI IPC + IEEE 802.15.4.
```

> The configuration can albo be configured to serve as an empty application when no radio serialization is used.

Reviewer: ```suggestion
The configuration can also serve as an empty application when no radio serialization is used.
```

> The firmware uses default overlays (found in the firmware directory) for specific radio configurations.

Reviewer: ```suggestion
The firmware uses default overlays for specific radio configurations.
The overlay files are located in the firmware directory.
```
